samples/config/config.json

```diff
@@ -0,0 +1,77 @@
+{
+  "enclave_file": "lib_jsgeneric.enclave.so.signed",
+  "enclave_type": "release",
+  "network": {
+    "node_address": { "hostname": "127.0.0.1", "port": "0" },
+    "rpc_interfaces": [
+      {
+        "rpc_address": { "hostname": "127.0.0.1", "port": "0" },
+        "public_rpc_address": { "hostname": "ccf.dummy.com", "port": "12345" },
+        "max_open_sessions_soft": 1000,
+        "max_open_sessions_hard": 1010
+      }
+    ]
+  },
+  "node_certificate": {
+    "subject_name": "CN=CCF Node",
+    "subject_alt_names": ["IPAddress:127.0.0.1", "dNSName:ccf.dummy.com"],
+    "curve_id": "secp384r1",
+    "initial_validity_period_days": 1
+  },
+  "start": {
+    "constitution_files": [],
+    "members": [
+      {
+        "certificate_file": "member0_cert.pem",
+        "encryption_public_key_file": "member0_enc_pubk.pem",
+        "data_json_file": ""
+      }
+    ],
+    "service_configuration": {
+      "recovery_threshold": 0,
+      "max_allowed_node_cert_validity_days": 365
+    }
+  },
+  "join": {
+    "join_timer_ms": 1000,
+    "target_rpc_address": { "hostname": "127.0.0.1", "port": "0" }
+  },
+  "ledger": {
+    "ledger_dir": "ledger",
+    "read_only_ledger_dirs": [],
+    "ledger_chunk_bytes": 20000
+  },
+  "snapshots": {
+    "snapshots_dir": "snapshots",
+    "snapshot_tx_interval": 10000
+  },
+  "logging": {
+    "host_log_level": "info",
+    "log_format_json": false
+  },
+  "consensus": {
+    "type": "CFT",
+    "timeout_ms": 100,
+    "election_timeout_ms": 4000
+  },
+  "intervals": {
+    "sig_tx_interval": 5000,
+    "sig_ms_interval": 1000,
+    "jwt_key_refresh_interval_s": 1800
+  },
+  "network_certificate_file": "networkcert.pem",
+  "node_certificate_file": "node.pem",
+  "node_pid_file": "node.pid",
+  "node_address_file": "node.node_address",
+  "rpc_addresses_file": "node.rpc_addresses",
+  "tick_period_ms": 1,
+  "io_logging_threshold_ns": 10000000,
+  "client_connection_timeout_ms": 2000,
+  "node_client_interface": null,
+  "worker_threads": 0,
+  "memory": {
+    "circuit_size_shift": 22,
+    "max_msg_size_shift": 24,
+    "max_fragment_size_shift": 16
+  }
```

Comment:
It would be nice to be able to express those as size directly. However, the numbers would be huge, so we should think of string sizes (e.g. used by CLI11) instead. In any case, this will be done in a separate PR.

Reply:
Follow-up in https://github.com/microsoft/CCF/issues/2612#issuecomment-973067527